src/kOS/Serialization/ConfigNodeFormatter.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using kOS.Safe.Serialization;
+using System.Collections.Generic;
+using kOS.Safe;
+using kOS.Safe.Utilities;
+using System.Linq;
+using System.Collections;
+using kOS.Safe.Encapsulation;
+
+namespace kOS.Serialization
+{
+    public class ConfigNodeFormatter : IFormatWriter, IFormatReader
+    {
+        private const string ParentNode = "";
+        private const string ListKey = "$list";
+        private const string BooleanKey = "$boolean";
+        private const string ScalarKey = "$scalar";
+        private const string ValueKey = "$value";
+        private static readonly ConfigNodeFormatter instance = new ConfigNodeFormatter();
+
+        public static ConfigNodeFormatter Instance
+        {
+            get 
+            {
+                return instance;
+            }
+        }
+
+        private ConfigNodeFormatter()
+        {
+        }
+
+        public ConfigNode ToConfigNode(Dump dump)
+        {
+            return ToConfigNode(ParentNode, dump);
+        }
+
+        private ConfigNode ToConfigNode(string name, List<object> list)
+        {
+            ConfigNode configNode = new ConfigNode();
+            configNode.name = name;
+
+            // ConfigNode doesn't handle arrays so we wrap everything in an extra node
+            // and give it a special name to mark this data set as an array
+            ConfigNode innerNode = new ConfigNode();
+            innerNode.name = ListKey;
+
+            var indexedList = list.Select((value, index) => new { Value = value, Index = index.ToString() });
+
+            foreach (var item in indexedList) {
+                HandleValue(innerNode, item.Index, item.Value);
+            }
```

Comment:
(See elsewhere in the PR comments for the comment with the word "curly brace" in big letters.)